src/main/java/org/zalando/nakadi/service/EventTypeService.java

```diff
@@ -184,6 +184,13 @@ public EventTypeService(final EventTypeRepository eventTypeRepository,
             final EventType eventType = schemaEvolutionService.evolve(original, eventTypeBase);
             eventType.setDefaultStatistic(
                     validateStatisticsUpdate(original.getDefaultStatistic(), eventType.getDefaultStatistic()));
+            final Long newRetentionTime = eventTypeBase.getOptions().getRetentionTime();
+            final Long oldRetentionTime = original.getOptions().getRetentionTime();
+            if (newRetentionTime != null && !newRetentionTime.equals(oldRetentionTime)) {
+                timelineService.getActiveTimelinesOrdered(eventTypeName)
+                        .forEach(timeline -> timelineService.getTopicRepository(timeline)
+                                .setRetentionTime(timeline.getTopic(), newRetentionTime));
```

Comment:
Here you don't catch `TopicConfigException`, and it's not in the method signature either (if I remember correctly, we had agreed to include unchecked exceptions in method signatures)